perfkitbenchmarker/providers/docker/docker_virtual_machine.py

```diff
@@ -0,0 +1,303 @@
+    inspect_cmd = ['docker', 'inspect', self.name]
+    info, _, returnCode = vm_util.IssueCommand(inspect_cmd, suppress_warning=True)
+    info = json.loads(info)
+    return info, returnCode
+
+  def _ConfigureProxy(self):
+    """
+    In Docker containers environment variables from /etc/environment
+    are not sourced - this results in connection problems when running
+    behind proxy. Prepending proxy environment variables to bashrc
+    solves the problem. Note: APPENDING to bashrc will not work because
+    the script exits when it is NOT executed in interactive shell.
+    """
+    if FLAGS.http_proxy:
+      http_proxy = "sed -i '1i export http_proxy=%s' /etc/bash.bashrc"
+      self.RemoteCommand(http_proxy % FLAGS.http_proxy)
+    if FLAGS.https_proxy:
+      https_proxy = "sed -i '1i export https_proxy=%s' /etc/bash.bashrc"
+      self.RemoteCommand(https_proxy % FLAGS.http_proxy)
+    if FLAGS.ftp_proxy:
+      ftp_proxy = "sed -i '1i export ftp_proxy=%s' /etc/bash.bashrc"
+      self.RemoteCommand(ftp_proxy % FLAGS.ftp_proxy)
+
+  def _BuildVolumesBody(self):
+    """
+    Constructs volumes-related part of create command for Docker Container
+    """
+    volumes = []
+
+    for scratch_disk in self.scratch_disks:
+      vol_string = scratch_disk.volume_name + ":" + scratch_disk.mount_point
+      volumes.append('-v')
+      volumes.append(vol_string)
+
+    return volumes
+
+  def _LocalImageExists(self, docker_image_name):
+    """
+    Checks if an image exists locally
+    Returns boolean
+    """
+    inspect_cmd = ['docker', 'image', 'inspect', docker_image_name]
+    info, _, returnCode = vm_util.IssueCommand(inspect_cmd, suppress_warning=True)
+    info = json.loads(info)
+    logging.info("Checking if Docker Image Exists")
+    if len(info) > 0 and returnCode == 0:
+      logging.info("Image exists")
+      return True
+    logging.info("Image does not exist")
+    return False
+
+  def _BuildImageLocally(self):
+    """
+    Build Container Image Locally
+    """
+
+    # Dockerfiles located at PerfKitBenchmarker/docker/<containerImage>/Dockerfile
+    directory = os.path.dirname(
+        data.ResourcePath(os.path.join('docker', self.container_image, 'Dockerfile')))
+    build_cmd = [
+        'docker', 'build', '--no-cache',
+        '-t', self.container_image, directory]
+
+    vm_util.IssueCommand(build_cmd)
+
+
+class DebianBasedDockerVirtualMachine(DockerVirtualMachine,
+                                      linux_virtual_machine.DebianMixin):
+  DEFAULT_IMAGE = UBUNTU_IMAGE
+
+  def ApplySysctlPersistent(self, sysctl_params):
+    """Override ApplySysctlPeristent function for Docker provider
+    Parent function causes errors with Docker because it shutdowns container
+    Args:
+      sysctl_params: dict - the keys and values to write
+    """
+    logging.warn("Sysctl flags NOT applied to Docker container. "
+                 "Please use --docker_sysctl_flags if sysctl need to be applied")
```

Comment:
Ah, this is the override I was thinking of above. Can you use the same flag and disable applying it here?

Reply:
switched to use --sysctl flag. removed --docker_sysctl_flags